homeassistant/components/water_heater/__init__.py

```diff
@@ -0,0 +1,327 @@
+"""
+Provides functionality to interact with water heater devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/water_heater/
+"""
+from datetime import timedelta
+import logging
+import functools as ft
+
+import voluptuous as vol
+
+from homeassistant.loader import bind_hass
+from homeassistant.helpers.temperature import display_temp as show_temp
+from homeassistant.util.temperature import convert as convert_temperature
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_TEMPERATURE, SERVICE_TURN_ON, SERVICE_TURN_OFF,
+    STATE_ON, STATE_OFF, STATE_UNKNOWN, TEMP_CELSIUS, PRECISION_WHOLE,
+    PRECISION_TENTHS, TEMP_FAHRENHEIT)
+
+DEFAULT_MIN_TEMP = 110
+DEFAULT_MAX_TEMP = 140
+
+DOMAIN = 'water_heater'
+
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+SCAN_INTERVAL = timedelta(seconds=60)
+
+SERVICE_SET_AWAY_MODE = 'set_away_mode'
+SERVICE_SET_TEMPERATURE = 'set_temperature'
+SERVICE_SET_OPERATION_MODE = 'set_operation_mode'
+
+STATE_ECO = 'eco'
+STATE_ELECTRIC = 'electric'
+STATE_PERFORMANCE = 'performance'
+STATE_HIGH_DEMAND = 'high_demand'
+STATE_HEAT_PUMP = 'heat_pump'
+STATE_GAS = 'gas'
+
+SUPPORT_TARGET_TEMPERATURE = 1
+SUPPORT_OPERATION_MODE = 2
+SUPPORT_AWAY_MODE = 4
+
+ATTR_MAX_TEMP = 'max_temp'
+ATTR_MIN_TEMP = 'min_temp'
+ATTR_AWAY_MODE = 'away_mode'
+ATTR_OPERATION_MODE = 'operation_mode'
+ATTR_OPERATION_LIST = 'operation_list'
+
+CONVERTIBLE_ATTRIBUTE = [
+    ATTR_TEMPERATURE,
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+ON_OFF_SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+SET_AWAY_MODE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Required(ATTR_AWAY_MODE): cv.boolean,
+})
+SET_TEMPERATURE_SCHEMA = vol.Schema(vol.All(
+    cv.has_at_least_one_key(
+        ATTR_TEMPERATURE),
+    {
+        vol.Exclusive(ATTR_TEMPERATURE, 'temperature'): vol.Coerce(float),
+        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Optional(ATTR_OPERATION_MODE): cv.string,
+    }
+))
+SET_OPERATION_MODE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Required(ATTR_OPERATION_MODE): cv.string,
+})
+
+
+@bind_hass
+def set_away_mode(hass, away_mode, entity_id=None):
```

Comment:
Drop these methods, we have just deleted them from the rest of the code.

Reply:
So do I need to create a common.py in the test like climate is doing or just remove these can call the services directly?